packages/typespec-ts/test/modularIntegration/generated/type/enum/extensible/src/api/string/index.ts

```diff
@@ -37,7 +37,7 @@ export async function _getKnownValueDeserialize(
     throw createRestError(result);
   }
 
-  return result.body;
+  return result.body as DaysOfWeekExtensibleEnum;
```

Comment:
in this case, it's exaustive enum and we didn't enable the experimental feature, RLC layer is generated as string and Modular layer is generated as literal unions. we have to use the cast here to make compiler happy.